daemon/daemon_unix.go

```diff
@@ -260,7 +260,7 @@ func getBlkioThrottleDevices(devs []*blkiodev.ThrottleDevice) ([]specs.LinuxThro
 
 	for _, d := range devs {
 		if err := unix.Stat(d.Path, &stat); err != nil {
-			return nil, err
+			return nil, errors.Wrapf(err, "failed to stat %q", d.Path)
```

Comment:
Wonder if we should replace `unix.Stat` for `os.Stat`. `os.Stat` returns a more useful error message;

https://play.golang.org/p/VYaEWtH5q4_O

```go
package main

import (
	"fmt"
	"os"

	"golang.org/x/sys/unix"
)

func main() {
	var stat unix.Stat_t
	err := unix.Stat("/no/such/path", &stat)
	fmt.Println(err)

	_, err = os.Stat("/no/such/path")
	fmt.Println(err)
}
```

Running that produces;

```
no such file or directory
stat /no/such/path: no such file or directory
```

Reply:
Ah, never mind that's not an option, because `os.Stat()` doesn't return the lower-level info. So, we could change it to a `os.PathError`, as was suggested on https://github.com/moby/moby/pull/41327/files#r467236003, and which is the same as `os.Stat` returns

```suggestion
			return nil, errors.WithStack(&os.PathError{Op: "stat", Path: d.Path, Err: err})
```

Reply:
Perfect solution. Thanks for you suggestion.

Reply:
Is it valid to return `os.PathError` from a non-os pkg?

Reply:
See https://github.com/moby/moby/pull/41327#discussion_r469892124

I checked the Golang source, and at least it doesn't call out that it's "exported but not for external use", so from that perspective, I think it should be ok;

```go
// Portable analogs of some common system call errors.
...
// PathError records an error and the operation and file path that caused it. 
```